common/main/cpp/native_c4multipeerreplicator.cc

```diff
@@ -0,0 +1,610 @@
+                    cls_C4MultipeerReplicator,
+                    "onDocumentEnded",
+                    "(J[BZ[Lcom/couchbase/lite/internal/core/C4DocumentEnded;)V");
+
+            if (m_C4MultipeerReplicator_onDocumentEnded == nullptr)
+                return false;
+        }
+
+        {
+            jclass localClass = env->FindClass("com/couchbase/lite/internal/MultipeerReplicationCollection");
+            if (localClass == nullptr)
+                return false;
+
+            cls_MultipeerReplColl = reinterpret_cast<jclass>(env->NewGlobalRef(localClass));
+            if (cls_MultipeerReplColl == nullptr)
+                return false;
+
+            f_MultipeerReplColl_token = env->GetFieldID(cls_MultipeerReplColl, "token", "J");
+            if (f_MultipeerReplColl_token == nullptr)
+                return false;
+
+            f_MultipeerReplColl_scope = env->GetFieldID(cls_MultipeerReplColl, "scope", "Ljava/lang/String;");
+            if (f_MultipeerReplColl_scope == nullptr)
+                return false;
+
+            f_MultipeerReplColl_name = env->GetFieldID(cls_MultipeerReplColl, "name", "Ljava/lang/String;");
+            if (f_MultipeerReplColl_name == nullptr)
+                return false;
+
+            f_MultipeerReplColl_options = env->GetFieldID(cls_MultipeerReplColl, "options", "[B");
+            if (f_MultipeerReplColl_options == nullptr)
+                return false;
+
+            f_MultipeerReplColl_pushFilter = env->GetFieldID(
+                    cls_MultipeerReplColl,
+                    "c4PushFilter",
+                    "Lcom/couchbase/lite/internal/MultipeerReplicationCollection$C4Filter;");
+            if (f_MultipeerReplColl_pushFilter == nullptr)
+                return false;
+
+            f_MultipeerReplColl_pullFilter = env->GetFieldID(
+                    cls_MultipeerReplColl,
+                    "c4PullFilter",
+                    "Lcom/couchbase/lite/internal/MultipeerReplicationCollection$C4Filter;");
+            if (f_MultipeerReplColl_pullFilter == nullptr)
+                return false;
+
+            m_MultipeerReplColl_filterCallback = env->GetStaticMethodID(
+                    cls_MultipeerReplColl,
+                    "filterCallback",
+                    "(J[BLjava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;IJZ)Z");
+            if (m_MultipeerReplColl_filterCallback == nullptr)
+                return false;
+        }
+
+        jniLog("multipeer replicator initialized");
+        return true;
+    }
+
+    //-------------------------------------------------------------------------
+    // Utility methods
+    //-------------------------------------------------------------------------
+
+    static int getC4CertChainSize(C4Cert *c4CertChain) {
+        int size = 0;
+        C4Cert *cert = c4CertChain;
+        while (cert != nullptr) {
+            size++;
+            cert = c4cert_nextInChain(cert);
```

Comment:
Need to release the cert, otherwise memory leak.

Reply:
Fixed